iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/LoadTsFileAnalyzer.java

```diff
@@ -250,7 +255,16 @@ private void analyzeSingleTsFile(final File tsFile, final boolean isDeleteAfterL
         tsFileResource.deserialize();
       }
 
-      schemaAutoCreatorAndVerifier.setCurrentModificationsAndTimeIndex(tsFileResource);
+      final TsFileDeviceIterator tsFileDeviceIterator = reader.getAllDevicesIteratorWithIsAligned();
+      final AtomicInteger deviceCount = new AtomicInteger();
+      tsFileDeviceIterator.forEachRemaining(o -> deviceCount.getAndIncrement());
```

Comment:
Please reduce the cost here when there is no mods file

Reply:
fixed